apps/web/pages/success.tsx

```diff
@@ -763,42 +761,62 @@ const getEventTypesFromDB = async (id: number) => {
   };
 };
 
-const strToNumber = z.string().transform((val, ctx) => {
-  const parsed = parseInt(val);
-  if (isNaN(parsed)) ctx.addIssue({ code: z.ZodIssueCode.custom, message: "Not a number" });
-  return parsed;
-});
-
 const schema = z.object({
-  type: strToNumber,
-  date: z.string().optional(),
-  username: z.string().optional(),
-  reschedule: z.string().optional(),
-  name: z.string().optional(),
-  email: z.string().optional(),
-  recur: z.string().optional(),
-  location: z.string().optional(),
-  eventSlug: z.string().default("15min"),
-  eventName: z.string().default(""),
-  bookingId: strToNumber,
+  uid: z.string(),
 });
 
 export async function getServerSideProps(context: GetServerSidePropsContext) {
   const ssr = await ssrInit(context);
   const parsedQuery = schema.safeParse(context.query);
   if (!parsedQuery.success) return { notFound: true };
-  const {
-    type: eventTypeId,
-    recur: recurringEventIdQuery,
-    eventSlug: eventTypeSlug,
-    eventName: dynamicEventName,
-    bookingId,
-    username,
-    name,
-    email,
-  } = parsedQuery.data;
-
-  const eventTypeRaw = !eventTypeId ? getDefaultEvent(eventTypeSlug) : await getEventTypesFromDB(eventTypeId);
+  const { uid } = parsedQuery.data;
+
+  const bookingInfo = await prisma.booking.findFirst({
+    where: {
+      uid,
+    },
+    select: {
+      title: true,
+      id: true,
+      uid: true,
+      description: true,
+      customInputs: true,
+      smsReminderNumber: true,
+      recurringEventId: true,
+      startTime: true,
+      location: true,
+      status: true,
+      user: {
+        select: {
+          id: true,
+          name: true,
+          email: true,
+          username: true,
+        },
+      },
+      attendees: {
+        select: {
+          name: true,
+          email: true,
+        },
+      },
+      eventTypeId: true,
+      eventType: true,
+    },
+  });
+
+  if (!bookingInfo || !bookingInfo.eventType?.slug)
+    return {
+      notFound: true,
+    };
+
+  // @NOTE: had to do this because Server side cant return [Object objects]
+  // probably fixable with json.stringify -> json.parse
+  bookingInfo["startTime"] = (bookingInfo?.startTime as Date)?.toISOString() as unknown as Date;
```

Comment:
Correct! Anything returned in getserversideprops should be serializable 👍  Perhaps towards the future we should convert this into something reusable, since I see we're actually doing this conversion (including the comment) in multiple places.

Reply:
I actually see that when you create custom events and try to book those, the `periodeStartDate` is not null and will throw a similar error. I think unfortunately we have to convert the `periodStartDate` and `periodEndDate` in the `eventType` object as well. Could you check and change this? 🙏  

![CleanShot 2022-11-01 at 08 47 25@2x](https://user-images.githubusercontent.com/2969573/199184424-85aad748-bd22-4e7a-a5e3-692866aabe46.jpg)